lib/sales_engine.rb

```diff
@@ -4,32 +4,47 @@
 require_relative 'invoice_repository'
 require_relative 'invoice_item_repository'
 require_relative 'transaction_repository'
+require_relative 'sales_analyst'
 
 class SalesEngine
-  attr_reader :items,
-              :merchants,
-              :invoices,
-              :invoice_items,
-              :transactions
-
-  def self.from_csv(paths)
-    # fix these names to make them more clear
-    ip         = paths[:items]
-    mp     = paths[:merchants]
-    inp      = paths[:invoices]
-    iip = paths[:invoice_items]
-    tp  = paths[:transactions]
-    cp     = paths[:customers]
-
-    SalesEngine.new(ip, mp, inp, iip, tp, cp)
-  end
-
-  def initialize(ip, mp, inp, iip, tp, cp)
-    # @customer_repository = CustomerRepository.new(@customer_path)
-    @transactions = TransactionRepository.new(tp)
-    @invoice_items = InvoiceItemRepository.new(iip)
-    @invoices = InvoiceRepository.new(inp)
-    @items = ItemRepository.new(ip)
-    @merchants = MerchantRepository.new(mp)
+  def self.from_csv(data)
+    SalesEngine.new(data)
+  end
+
+  def initialize(data)
+    @items          = data[:items]
+    @merchants      = data[:merchants]
+    @invoices       = data[:invoices]
+    @invoice_items  = data[:invoice_items]
+    @transactions   = data[:transactions]
+    @customers      = data[:customers]
+  end
+
+  def items
```

Comment:
am I correct to assume that @items and method items, etc. are two different things?  if so, should we call them different things to avoid confusion? I know you don't like messy, but maybe we can just put a 'd' for data or something? I am guessing it'd be ok since it will be called differently, but it's also the kind of thing I might do.

Reply:
otherwise, it looks soooo much cleaner.  Thanks, Sam!